lib/web_ui/lib/src/engine/semantics/tabs.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'label_and_value.dart';
+import 'semantics.dart';
+
+/// Represents semantic objects that deliver information in a visual manner.
+///
+/// Uses aria tab role to convey this semantic information to the element.
+///
+/// Screen-readers takes advantage of "aria-label" to describe the visual.
+class SemanticTab extends SemanticRole {
+  SemanticTab(SemanticsObject semanticsObject)
+      : super.withBasics(
+          SemanticRoleKind.tab,
+          semanticsObject,
+          preferredLabelRepresentation: LabelRepresentation.ariaLabel,
+        );
+  @override
+  bool focusAsRouteDefault() => focusable?.focusAsRouteDefault() ?? false;
+
+  @override
+  void update() {
+    super.update();
+    setAriaRole('tab');
+  }
+}
+
+/// Represents semantic objects that deliver information in a visual manner.
+///
+/// Uses aria tab role to convey this semantic information to the element.
+///
+/// Screen-readers takes advantage of "aria-label" to describe the visual.
+class SemanticTabPanel extends SemanticRole {
+  SemanticTabPanel(SemanticsObject semanticsObject)
+      : super.withBasics(
```

Comment:
`withBasics` will attempt to add the following behaviors:

```
    addFocusManagement();
    addLiveRegion();
    addRouteName();
    addLabelAndValue(preferredRepresentation: preferredLabelRepresentation);
    addSelectableBehavior();
```

I wonder if tab panels should at most have a label and nothing else. So we could use the `super.blank` constructor instead, and only `addLabelAndValue()`.

Reply:
Is this something we should enforce? developer can still assign onTap or another semantics action to it. If we do want to enforce, we should also enforce constraint on the framework side

Reply:
I guess let's start with `withBasics` and see. Can't tell off the top of my head if this is problematic at all.